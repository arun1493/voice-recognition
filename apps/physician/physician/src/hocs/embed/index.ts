export * from "./withEmbedAuth";
export * from "./withAmwellAuth";
