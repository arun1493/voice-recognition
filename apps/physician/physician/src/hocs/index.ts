export * from "./withIntercom";
export * from "./withAnalytics";
export * from "./withTestSetup";
export * from "./withBrowserSupport";
