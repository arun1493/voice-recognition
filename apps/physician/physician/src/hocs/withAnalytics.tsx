import { useLocation } from "react-router-dom";
import { FunctionComponent, useEffect } from "react";
import ReactGA, { FieldsObject, InitializeOptions } from "react-ga";

import { isDevelopmentEnvironment } from "../helpers/environment";

const ANALYTICS_TRACKING_ID = "UA-100205469-2";

const initializeAnalytics = (options: InitializeOptions = {}) => {
  ReactGA.initialize(ANALYTICS_TRACKING_ID, {
    debug: isDevelopmentEnvironment(),
    ...options
  });
};

function withAnalytics(WrappedComponent: FunctionComponent, options: FieldsObject = {}): FunctionComponent {
  return () => {
    const location = useLocation();

    const trackPage = () => {
      ReactGA.set({
        page: location.pathname,
        // TODO: Get userId from Redux
        userId: undefined,
        ...options
      });

      ReactGA.pageview(location.pathname);
    };

    useEffect(() => {
      trackPage();
    }, [location.pathname]);

    return <WrappedComponent />;
  };
}

export { initializeAnalytics, withAnalytics };
