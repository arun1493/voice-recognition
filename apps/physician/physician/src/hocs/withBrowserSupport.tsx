import _some from "lodash/some";
import { UAParser } from "ua-parser-js";
import { Paths } from "@suki-web/suki-business";
import { FunctionComponent, useEffect, useState } from "react";

import { getRoute } from "../helpers";
import { DEVICE_TYPES, Pages, SUPPORTED_BROWSERS } from "../constants";

function isSupportedBrowser(browserName: string, version: number | string): boolean {
  if (!(browserName in SUPPORTED_BROWSERS)) return false;
  if (typeof version === "string") {
    version = Number.parseInt(version);
  }

  return version >= SUPPORTED_BROWSERS[browserName];
}

function redirectUnsupportedBrowser(): void {
  const redirectRoute = getRoute(Pages.UnSupportedDeviceOrBrowser);
  if (redirectRoute) {
    window.location.href = redirectRoute;
  }
}

function getDeviceType() {
  const ua = navigator.userAgent;
  if (/(tablet|ipad|playbook|silk)|(android(?!.*mobi))/i.test(ua)) {
    return DEVICE_TYPES.tablet;
  }
  if (/Mobile|iP(hone|od)|Android|BlackBerry|IEMobile|Kindle|Silk-Accelerated|(hpw|web)OS|Opera M(obi|ini)/.test(ua)) {
    return DEVICE_TYPES.mobile;
  }
  return DEVICE_TYPES.desktop;
}

function isDesktop() {
  return getDeviceType() === DEVICE_TYPES.desktop;
}

function withBrowserSupport(WrappedComponent: FunctionComponent) {
  return () => {
    const [isBrowserSupported, setIsBrowserSupported] = useState<boolean>(false);

    const UNSUPPORTED_ROUTES: string[] = [
      Paths.AltaisEntry,
      Paths.Authentication,
      Paths.Embed,
      Paths.ForgotPassword,
      Paths.Help,
      Paths.Home,
      Paths.ImplicitCallback,
      Paths.IntroPage,
      Paths.Login,
      Paths.Patients,
      Paths.ScheduleImport,
      Paths.Settings,
      Paths.Settings,
      Paths.UnfinishedNotes,
      Paths.WindowsUpdate
    ];

    const isCurrentRouteUnSupported = (route: string) => {
      return window.location.pathname.includes(route);
    };

    const isUnsupportedRoute = _some(UNSUPPORTED_ROUTES, isCurrentRouteUnSupported);

    useEffect(() => {
      const parser = new UAParser();
      const browserInformation = parser.getBrowser();

      if (
        !isSupportedBrowser(browserInformation.name as string, browserInformation.version as string) ||
        (!isDesktop() && (isUnsupportedRoute || window.location.pathname === Paths.Root))
      ) {
        return redirectUnsupportedBrowser();
      }

      setIsBrowserSupported(true);
    }, []);

    return isBrowserSupported ? <WrappedComponent /> : null;
  };
}

export { withBrowserSupport };
