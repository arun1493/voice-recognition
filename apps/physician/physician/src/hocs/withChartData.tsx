import store from "store2";
import _isEmpty from "lodash/isEmpty";
import { FunctionComponent } from "react";
import { useDispatch } from "react-redux";
import { useQuery } from "@tanstack/react-query";
import { useLocation, useParams } from "react-router-dom";
import { EMPTY_OBJECT, Paths } from "@suki-web/suki-business";

import {
  getPatientAllergies,
  getPatientMedicalHistory,
  getPatientMedications,
  getPatientSurgicalHistory,
  getPatientVitals,
  logger
} from "../api";
import { noteActions } from "../redux/note";
import { getCardQueryParams } from "../helpers";
import { compositionReader, metadataReader } from "../readers";
import { CardType, LOCAL_STORE, QUERY_NAMES } from "../constants";

function withChartData(WrappedComponent: FunctionComponent<any>): FunctionComponent {
  return (props: any) => {
    const { composition, loading } = props;

    const dispatch = useDispatch();
    const location = useLocation();
    const isPatientProfilePage = location.pathname.includes(Paths.PatientProfile);
    const isEmrUser = store.get(LOCAL_STORE.IS_EMR_USER);

    let shouldFetchCardsData;
    let queryParams = EMPTY_OBJECT as any;

    if (isPatientProfilePage) {
      const { patientId = "" } = useParams<{ patientId: string }>();
      shouldFetchCardsData = !!(isEmrUser && patientId);
      queryParams = {
        sukiPatientId: patientId
      };
    } else {
      const metadata = compositionReader.metadata(composition);
      const patientInfo = metadataReader.patient(metadata);
      shouldFetchCardsData = (isEmrUser && !loading && !_isEmpty(composition) && !_isEmpty(patientInfo)) || false;
      queryParams = getCardQueryParams(composition);
    }

    // Allergies
    useQuery([QUERY_NAMES.PATIENT_ALLERGIES, queryParams], () => getPatientAllergies(queryParams), {
      enabled: shouldFetchCardsData,
      staleTime: 0,
      onError: (error) => logger.error({ message: "Error fetching allergies", payload: error }),
      onSuccess: (allergies) =>
        dispatch(noteActions.setCardsByType({ type: CardType.ALLERGIES, cards: allergies?.data || [] }))
    });

    // Medical History
    useQuery([QUERY_NAMES.PATIENT_MEDICAL_HISTORY, queryParams], () => getPatientMedicalHistory(queryParams), {
      enabled: shouldFetchCardsData,
      staleTime: 0,
      onError: (error) => logger.error({ message: "Error fetching medical history", payload: error }),
      onSuccess: (medicalHistory) =>
        dispatch(noteActions.setCardsByType({ type: CardType.PAST_MEDICAL_HISTORY, cards: medicalHistory?.data }))
    });

    // Medications
    useQuery([QUERY_NAMES.PATIENT_MEDICATIONS, queryParams], () => getPatientMedications(queryParams), {
      enabled: shouldFetchCardsData,
      staleTime: 0,
      onError: (error) => logger.error({ message: "Error fetching medications", payload: error }),
      onSuccess: (medications) =>
        dispatch(noteActions.setCardsByType({ type: CardType.CURRENT_MEDICATION, cards: medications?.data }))
    });

    // Surgical History
    useQuery([QUERY_NAMES.PATIENT_SURGICAL_HISTORY, queryParams], () => getPatientSurgicalHistory(queryParams), {
      enabled: shouldFetchCardsData,
      staleTime: 0,
      onError: (error) => logger.error({ message: "Error fetching surgical history", payload: error }),
      onSuccess: (surgicalHistory) =>
        dispatch(noteActions.setCardsByType({ type: CardType.PAST_SURGICAL_HISTORY, cards: surgicalHistory?.data }))
    });

    // Vitals
    useQuery([QUERY_NAMES.PATIENT_VITALS, queryParams], () => getPatientVitals(queryParams), {
      enabled: shouldFetchCardsData,
      staleTime: 0,
      onError: (error) => logger.error({ message: "Error fetching vitals", payload: error }),
      onSuccess: (vitals) => dispatch(noteActions.setCardsByType({ type: CardType.VITALS, cards: vitals?.data }))
    });

    return <WrappedComponent {...props} />;
  };
}

export { withChartData };
