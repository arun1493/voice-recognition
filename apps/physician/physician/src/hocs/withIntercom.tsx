import { FunctionComponent } from "react";
import { IntercomProvider } from "react-use-intercom";

import { isProductionEnvironment } from "../helpers/environment";

const INTERCOM_ID = isProductionEnvironment() ? "mrausu9h" : "tfrr9ikc";

function withIntercom(WrappedComponent: FunctionComponent): FunctionComponent {
  return () => {
    return (
      <IntercomProvider appId={INTERCOM_ID} autoBoot={true} autoBootProps={{ hideDefaultLauncher: true }}>
        <WrappedComponent />
      </IntercomProvider>
    );
  };
}

export { withIntercom };
