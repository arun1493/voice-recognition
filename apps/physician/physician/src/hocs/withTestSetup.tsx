import { Provider } from "react-redux";
import { lightTheme } from "@suki-web/suki-ui";
import { render } from "@suki-web/suki-business";
import { ThemeProvider } from "styled-components";
import { FunctionComponent, ReactNode } from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";

import { setupStore } from "../redux/store";

const queryClient = new QueryClient();

const componentSetup =
  (Component: FunctionComponent<any>, defaultProps = {}) =>
  (overriddenProps = {}) =>
    render(
      <Router>
        <ThemeProvider theme={lightTheme}>
          <QueryClientProvider client={queryClient}>
            <Component {...defaultProps} {...overriddenProps} />
          </QueryClientProvider>
        </ThemeProvider>
      </Router>
    );

const componentSetupWithStore =
  (getJSX: (defaultProps: any) => ReactNode) =>
  (overriddenProps = {}, preloadedState = {}) =>
    render(
      <Provider store={setupStore(preloadedState)}>
        <QueryClientProvider client={queryClient}>
          <Router>
            <ThemeProvider theme={lightTheme}>{getJSX(overriddenProps)}</ThemeProvider>
          </Router>
        </QueryClientProvider>
      </Provider>
    );

const componentWithTheme = (Component: FunctionComponent<any>, props = {}) => (
  <ThemeProvider theme={lightTheme}>
    <Component {...props} />
  </ThemeProvider>
);

const componentWithThemeAndStore = (Component: FunctionComponent<any>, props = {}, preloadedState = {}) => (
  <Provider store={setupStore(preloadedState)}>
    <QueryClientProvider client={queryClient}>
      <Router>
        <ThemeProvider theme={lightTheme}>
          <Component {...props} />
        </ThemeProvider>
      </Router>
    </QueryClientProvider>
  </Provider>
);

export { componentSetup, componentSetupWithStore, componentWithTheme, componentWithThemeAndStore };
