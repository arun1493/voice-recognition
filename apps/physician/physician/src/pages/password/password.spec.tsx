import "@testing-library/jest-dom/extend-expect";
import { cleanup, RenderResult } from "@testing-library/react";

import { Password } from ".";
import { componentSetupWithStore } from "../../hocs";
import { shouldMatchSnapshot } from "../../helpers/snapshot.spec";

const setup = componentSetupWithStore(() => <Password />);

describe("Password component", () => {
  let component: RenderResult;

  beforeEach(() => {
    component = setup();
  });

  afterEach(() => {
    cleanup();
  });

  it("should render successfully", () => {
    const { baseElement } = component;
    expect(baseElement).toBeTruthy();
  });

  it("should match snapshot", () => {
    shouldMatchSnapshot(Password);
  });
});
