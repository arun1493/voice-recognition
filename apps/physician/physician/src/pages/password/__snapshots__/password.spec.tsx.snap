// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password component should match snapshot 1`] = `
<div
  className="passwordstyled__StyledPage-sc-1v4h3ir-0 ksfocP"
>
  <div
    className="passwordstyled__StyledHeader-sc-1v4h3ir-1 ceNaZe"
  >
    <a
      className="sukiLogoRstyled__StyledSukiLogoR-njr1sc-0 evdjAv"
      data-cy="suki-logo"
      href="/https://suki.ai"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 82 24"
        width="82"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clipPath="url(#clip0_297_94)"
        >
          <path
            d="M0 8.36132C0 3.91566 2.87942 1.31323 10.8235 1.31323C15.4949 1.31323 18.6239 2.12371 19.9679 2.93419V7.49271C18.429 6.8498 14.7117 5.87175 10.5704 5.87175C7.57815 5.87175 5.34164 6.4873 5.34164 8.19375C5.34164 9.90021 6.54539 10.1225 10.1806 10.3482C18.0391 10.8235 20.5834 12.1675 20.5834 16.5277C20.5834 21.5068 16.808 23.3501 9.73258 23.3501C5.84434 23.3501 2.06894 22.6216 0.253061 21.7565V17.3381C1.8193 18.0084 5.34164 18.9044 9.37009 18.9044C13.3985 18.9044 15.2144 18.2068 15.2144 16.6097C15.2144 15.406 14.4587 14.9033 10.013 14.6229C2.82471 14.1783 0 12.7796 0 8.36132Z"
            fill="white"
          />
          <path
            d="M23.3124 6.55225H28.4591V15.9668C28.4591 18.2033 29.6628 18.9591 32.0122 18.9591C34.3616 18.9591 36.1501 18.2888 37.6035 17.4783V6.55225H42.7194V22.9567H37.8531V21.0554C36.0646 22.2283 33.6879 23.35 30.4973 23.35C25.5489 23.35 23.309 21.1409 23.309 17.3655V6.55225H23.3124Z"
            fill="white"
          />
          <path
            d="M46.4504 22.9601V0.960938H51.5663V13.1147L58.9222 6.55222H65.8301L57.2157 13.4498L66.3601 22.9567H59.3975L53.7481 16.3566L51.5663 18.2033V22.9567H46.4504V22.9601Z"
            fill="white"
          />
          <path
            d="M68.3093 2.34936C68.3093 0.783121 69.26 0 71.2742 0C73.2885 0 74.2665 0.783121 74.2665 2.34936C74.2665 3.91561 73.3158 4.67137 71.2742 4.67137C69.2327 4.67137 68.3093 3.88825 68.3093 2.34936Z"
            fill="white"
          />
          <path
            d="M73.8459 6.55225H68.73V22.9567H73.8459V6.55225Z"
            fill="white"
          />
          <path
            d="M80.0357 9.83854C79.0474 9.83854 78.2369 9.02122 78.2369 8.03976C78.2369 7.05829 79.0474 6.23413 80.0357 6.23413C81.024 6.23413 81.8344 7.02409 81.8344 8.03976C81.8344 9.05542 81.0342 9.83854 80.0357 9.83854ZM80.0357 6.62056C79.2457 6.62056 78.6336 7.26005 78.6336 8.03976C78.6336 8.81946 79.2457 9.44185 80.0357 9.44185C80.8256 9.44185 81.448 8.83997 81.448 8.03976C81.448 7.23954 80.829 6.62056 80.0357 6.62056ZM80.0801 8.32017L80.3947 8.96651H80.8085L80.4495 8.26546C80.6546 8.18338 80.8085 7.98504 80.8085 7.7183C80.8085 7.35923 80.5315 7.12326 80.1793 7.12326H79.3688V8.96651H79.7279V8.32017H80.0801ZM80.4392 7.72514C80.4392 7.91322 80.3229 8.03976 80.1075 8.03976H79.7313V7.43788H80.1075C80.3127 7.43788 80.4392 7.54731 80.4392 7.72514Z"
            fill="white"
          />
        </g>
        <defs>
          <clipPath
            id="clip0_297_94"
          >
            <rect
              fill="white"
              height="23.35"
              width="81.8344"
            />
          </clipPath>
        </defs>
      </svg>
    </a>
  </div>
  <div
    className="passwordstyled__StyledContainer-sc-1v4h3ir-2 lcqDyo"
  >
    <h5
      className="headerstyled__StyledH5-sc-1y4xvz5-4 iITAex passwordstyled__StyledTitle-sc-1v4h3ir-3 eJWHzR"
    >
      pages.password.title
    </h5>
    <div
      className="passwordstyled__StyledContent-sc-1v4h3ir-4 dWOVlx"
    >
      <svg
        className="passwordstyled__StyledIcon-sc-1v4h3ir-7 WNHWu"
        fill="none"
        height={42}
        viewBox="0 0 42 42"
        width={42}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M34.142 5.858C30.3913 2.10667 25.304 0 20 0C8.954 0 0 8.954 0 20C0 25.304 2.10667 30.3913 5.858 34.142C9.60867 37.8933 14.696 40 20 40C25.304 40 30.3913 37.8933 34.142 34.142C37.8933 30.3913 40 25.304 40 20C40 14.696 37.8933 9.60867 34.142 5.858ZM26.048 10.4656C26.8088 10.7772 27.4989 11.2427 28.1183 11.8621C28.7285 12.4723 29.1871 13.16 29.4941 13.9254C29.8057 14.6862 29.9608 15.4694 29.9596 16.2749C29.9676 17.0805 29.819 17.8623 29.5138 18.6206C29.2178 19.3788 28.7672 20.0605 28.1622 20.6655C27.2225 21.6052 26.0879 22.162 24.7584 22.3359C23.4335 22.5145 22.024 22.2784 20.5299 21.6277L20.5316 25.3285C20.5322 25.4758 20.4867 25.5953 20.395 25.687C20.3079 25.7833 20.1954 25.8312 20.0573 25.8306L16.3357 25.808L16.3677 28.5078C16.3637 28.6505 16.3159 28.7677 16.2242 28.8594C16.1371 28.9557 16.02 29.0035 15.8727 29.0029L11.4952 28.9845C11.3479 28.9839 11.2279 28.9281 11.1354 28.8173C11.0475 28.711 11.0033 28.5981 11.0028 28.4784L11 26.1654C11.0041 26.0411 11.0243 25.9284 11.0607 25.8273C11.1017 25.7308 11.1658 25.639 11.2529 25.5519L18.0391 18.7657C17.6813 17.9679 17.4938 17.1524 17.4765 16.3192C17.4638 15.4814 17.61 14.6741 17.9151 13.8975C18.2295 13.1209 18.6938 12.4255 19.308 11.8113C19.9176 11.2017 20.5992 10.7465 21.3529 10.4459C22.1157 10.1453 22.8976 9.99669 23.6985 10.0001C24.5087 10.0035 25.2918 10.1586 26.048 10.4656ZM23.4012 14.832C23.4032 15.3107 23.5753 15.7211 23.9173 16.0631C24.2547 16.4006 24.6628 16.5703 25.1415 16.5723C25.6249 16.5789 26.0362 16.4126 26.3754 16.0734C26.71 15.7388 26.874 15.3298 26.8673 14.8465C26.8699 14.3632 26.7025 13.9528 26.365 13.6154C26.0184 13.2687 25.6057 13.0944 25.1269 13.0924C24.6528 13.095 24.2462 13.2659 23.907 13.6051C23.5678 13.9443 23.3992 14.3532 23.4012 14.832Z"
          fill="#2E5F62"
          fillRule="evenodd"
        />
      </svg>
      <div>
        <input
          className="passwordstyled__StyledFormInput-sc-1v4h3ir-5 dCNELH"
          height="40px"
          onChange={[Function]}
          placeholder="pages.password.placeholder1"
          type="password"
          value=""
        />
        <p
          className="textstyled__StyledText-sc-1vedkxi-0 gQoCmW passwordstyled__StyledText-sc-1v4h3ir-8 dPJBpL"
          color="inherit"
          cursor="default"
          onClick={[Function]}
          size="md"
        >
          pages.password.requirements
        </p>
        <ul>
          <li>
            pages.password.requirement1
          </li>
          <li>
            pages.password.requirement2
          </li>
          <li>
            pages.password.requirement3
          </li>
          <li>
            pages.password.requirement4
          </li>
          <li>
            pages.password.requirement5
          </li>
        </ul>
        <input
          className="passwordstyled__StyledFormInput-sc-1v4h3ir-5 dCNELH"
          height="40px"
          onChange={[Function]}
          placeholder="pages.password.placeholder2"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      className="passwordstyled__StyledSeparator-sc-1v4h3ir-6 gErYcP"
    />
    <button
      className="buttonstyled__StyledButton-ws70ng-0 eQJEpp"
      data-cy="button"
      disabled={false}
      height="56px"
      onClick={[Function]}
      type="button"
    >
      <p
        className="textstyled__StyledText-sc-1vedkxi-0 kTwEkJ"
        color="inherit"
        cursor="default"
        onClick={[Function]}
        size="xl"
      >
        pages.password.submit
      </p>
    </button>
  </div>
</div>
`;
